mangaki/mangaki/utils/mal.py

```diff
@@ -31,149 +40,256 @@ def random_ip():
-    except ET.ParseError:
-        print(HEADERS)
-        pass
+            try:
+                mal_entry = MALEntry(entry, work_type)
+                entries.append(mal_entry)
+            except ET.ParseError:
+                logger.exception('MAL parsing error during search')
+
+            return entries
+
+
+client = MALClient()
 
-    return entries
 
+def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry]):
+    # Assumption: Mangaki's slugs are the same as MAL's work types.
+    if not mal_entries:
+        return
 
-def poster_url(query):
+    # All entries are supposed of the same type.
+    work_cat = Category.objects.get(slug=mal_entries[0].work_type.value)
+    for entry in mal_entries:
+        is_present = Work.objects.filter(
+            category=work_cat,
+            ext_poster=entry.poster
+        ).exists()
+
+        if not is_present:
+            Work.objects.create(
+                category=work_cat,
+                title=entry.english_title or entry.title,
+                source=entry.source_url,
+                ext_poster=entry.poster,
+                date=entry.start_date
+            )
+
+
+def poster_url(work_type: MALWorks, query: str) -> str:
     """
     Look for an anime on MyAnimeList and return the URL of its poster.
     """
 
-    SEARCH_URL = 'http://myanimelist.net/api/anime/search.xml'
-    HEADERS = {
-        'X-Real-IP': random_ip(),
-        'User-Agent': 'Mozilla/5.0 (X11; Linux i686 on x86_64; rv:36.0) Gecko/20100101 Firefox/36.0'
-    }
+    return client.search_work(work_type, query).poster
 
-    r = requests.get(SEARCH_URL,
-                     params={'q': query},
-                     headers=HEADERS,
-                     auth=(settings.MAL_USER, settings.MAL_PASS))
 
-    html_code = html.unescape(re.sub(r'&amp;([A-Za-z]+);', r'&\1;', r.text))
-    xml = re.sub(r'&([^alg])', r'&amp;\1', _encoding_translation(html_code))
-    return ET.fromstring(xml).find('entry').find('image').text
+def compute_rating_choice_from_mal_score(score: int) -> Optional[str]:
+    """
+    Compute from the MAL score, "the Mangaki choice" (like, neutral, favorite, dislike).
+    :param score: an integer between 0 and 10
+    :type score: int
+    :return: None if the score is out of the range, otherwise, [7, 10] => like, [5, 6] => neutral, [0, 4] => dislike.
+    :rtype: Optional[str]
+    """
+    if 7 <= score <= 10:
+        return 'like'
+    elif 5 <= score <= 6:
+        return 'neutral'
+    elif score >= 0:
+        return 'dislike'
+    else:
+        return None
 
 
-def import_mal(mal_username, mangaki_username):
+def get_or_create_from_mal(work_list: QuerySet,
+                           work_type: MALWorks,
+                           title: str,
+                           poster_link: str) -> Optional[Work]:
+    """
+    Get a work from the current `work_list` (a queryset, filtered by category)
+    or create from scratch using MAL as reference.
```

Comment:
Please create an issue calling for a `Reference` check. (Add reference if does not exist.)